### PR TITLE
set up CI on fibonacci project [2.75]

### PR DESCRIPTION
**Pull request by ephieo**

### CODE REVIEW

File: spec/main_spec.rb [hunk 1]
```diff
@@ -4,14 +4,18 @@
...
 require_relative '../lib/main'
 
-describe Fibonacci do 
-    subject(:fibonacci) {described_class.new}
+describe Puzzle do 
+    subject(:puzzle) {described_class.new}
 
     it ' has a class called fibonacci' do
-        expect(fibonacci).to be_instance_of(Fibonacci)
+        expect(puzzle).to be_instance_of(Puzzle)
```
Comment (ellehallal) [inline spec/main_spec.rb:11]: I think we can remove this test

File: lib/main.rb [hunk 2]
```diff
@@ -1,3 +1,19 @@
...
+        end
+        
+        puts start_arr
+        return start_arr
+
+    end
+
+end 
+
+puzzle = Puzzle.new
```
Comment (ellehallal) [inline lib/main.rb:17]: What are your thoughts on moving this to a separate file, or alternatively moving the class to it's own file?

File: .github/workflows/ruby.yml [hunk 3]
```diff
@@ -1,35 +1,22 @@
...
-# documentation.
-# This workflow will download a prebuilt Ruby version, install dependencies and run tests with Rake
-# For more information see: https://github.com/marketplace/actions/setup-ruby-jruby-and-truffleruby
-
 name: Ruby
 
 on:
   push:
-    branches: [ main ]
+    branches: [main]
```
Comment (ellehallal) [inline .github/workflows/ruby.yml:5]: I'm not too familiar with github workflows. Does this mean, the pipeline can be run on all branches except main?

If yes, would we want to run the pipeline on main too, after merging a PR for example?

File: spec/main_spec.rb [hunk 4]
```diff
@@ -4,14 +4,18 @@
...
     end
 
-    it 'takes an array as an argument and returns third-to-last and second-to-last numbers'do 
-       new_arr
+    it 'it should take an integer and return the fibonacci sequence up to that integer'do
```
Comment (ellehallal) [inline spec/main_spec.rb:14]: I'm new to the fibonacci kata, and this test helped me to understand what it is, so thanks! :D

File: spec/main_spec.rb (hunk 1 above)
Comment (ephieo) [reply spec/main_spec.rb:11]: Thank you will do.

File: lib/main.rb (hunk 2 above)
Comment (ephieo) [reply lib/main.rb:17]: Yes, I think it's a good idea, I was thinking about extracting everything out into their own files.

File: .github/workflows/ruby.yml (hunk 3 above)
Comment (ephieo) [reply .github/workflows/ruby.yml:5]: I'm not too sure but I think it means that it runs on main only when merge into main or pushed up to main.

File: spec/main_spec.rb (hunk 4 above)
Comment (ephieo) [reply spec/main_spec.rb:14]: Thank you 😄

**Pull request merged**